src/main/java/com/box/sdk/BoxSignRequest.java

```diff
@@ -0,0 +1,153 @@
+package com.box.sdk;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.net.URL;
+import java.util.List;
+
+/**
+ *
+ */
+
+@BoxResourceType("sign_request")
+public class BoxSignRequest extends BoxResource{
+
+	public static final URLTemplate SIGN_REQUESTS_URL_TEMPLATE = new URLTemplate("sign_requests");
+
+	public static final URLTemplate SIGN_REQUEST_URL_TEMPLATE = new URLTemplate("sign_requests/%s");
+
+	public static final URLTemplate SIGN_REQUEST_CANCEL_URL_TEMPLATE = new URLTemplate("sign_requests/%s/cancel");
+
+	public static final URLTemplate SIGN_REQUEST_RESEND_URL_TEMPLATE = new URLTemplate("sign_requests/%s/resend");
+
+	public boolean isDocumentPreparationNeeded;
+	public String redirectUrl;
+	public String declinedRedirectUrl;
+	//public List<RequiredAttachment> required_attachments;
+	public boolean areAttachmentsEnabled;
+	public boolean areTextSignaturesEnabled;
+	public boolean isTextEnabled;
+	public boolean areDatesEnabled;
+	public boolean areEmailsDisabled;
+	public String signature_color;
+	public boolean isPhoneVerificationRequiredToView;
+	public String emailSubject;
+	public String emailMessage;
+	public boolean areRemindersEnabled;
+	//public List<Signer> signers;
+	//public List<SourceFile> source_files;
+	public List<BoxFile> source_files;
+	//public List<BoxSignRequestSigner> signers;
+	//public ParentFolder parent_folder;
+	public String name;
+	//public List<PrefillTag> prefill_tags;
+	public int daysValid;
+	//public String external_id;
+	//public String type;
+	//public String id;
+	public String prepareUrl;
+	//public SigningLog signing_log;
+	public String status;
+	//public SignFiles sign_files;
+	//public Date auto_expire_at;
+	//public Date created_at;
+	//public Date updated_at;
+
+	/**
+	 * Constructs a BoxResource for a resource with a given ID.
+	 *
+	 * @param api the API connection to be used by the resource.
+	 * @param id  the ID of the resource.
+	 */
+	public BoxSignRequest(BoxAPIConnection api, String id) {
+		super(api, id);
+	}
+
+	//TODO we can use BoxFile.Info instead if we want
```

Comment:
Yes, this is typical approach in our Java SDK, pretty handy for objects with a lot of fields. I also left a TODO comment regarding BoxFile.Info (which is an existing class representing File). For now I am using a DTO to pass a file id and file_version, but maybe we could expose also a method that takes BoxFile.Info instead - this way client could reuse existing model from the API instead of creating a DTO.